src/test/java/com/google/sps/BooksTest.java

```diff
@@ -0,0 +1,54 @@
+package com.google.sps.servlets;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.net.URL;
+import org.mockito.Mockito;
+import java.io.IOException;
+import java.util.Map;
+import java.util.HashMap;
+
+/** */
+@RunWith(JUnit4.class)
+public final class BooksTest {
+  
+  public String EMPTY_MAP_STRING = "{}";
+  
+ /**
+  * Checks that if the keyword is gibberish, no result is returned.
+  */
+  @Test
+  public void noOutputTest() throws IOException {
+    Map<String, String[]> badParams = new HashMap<>();
+    String[] gibberish = {"kjdaflkdsalklkjfdsalkflkds"};
+    badParams.put("key", gibberish);
+    ServletSetUpObject booksGet = TestingUtil.mockGet(badParams);
+    BooksServlet booksServlet = new BooksServlet();
+    booksServlet.doGet(booksGet.getRequest(), booksGet.getResponse());
+    String output = booksGet.getWriter().toString();
+    output = TestingUtil.getInnerMap(output);
+    Assert.assertEquals(output, EMPTY_MAP_STRING);
```

Comment:
I'm confused as to why test case returns an EMPTY_MAP_STRING and the other doesn't.

Reply:
For this test, my intention was to have a bizarre keyword that would not generate a result so the returned map is empty.

Reply:
I don't understand how its working though. You're not actually making a call to the API correct?

Reply:
In this case, the servlet would make a call to the api.

Reply:
Oh, ok. Usually when we make "unit test" we don't make calls to external APIs. I think you should reach out to Dan to get his opinion on this, and perhaps the VSE's.

Reply:
That makes sense. My thought with this was to show some example tests that used the entire servlet because I was working on an older branch and the specific methods in the servlets classes may be pretty different than those methods in later versions. I was planning on adding more tests for specific methods once this PR was merged and the code in the servlets is more stable. Let me know what you think of this plan.

Reply:
I just spoke with Daniel, and he also encouraged to do unit testing instead of integration testing (i.e. testing with API calls). I'll change the tests to reflect this and re-request a review after.

Reply:
SG, thanks for following up with him.